datafusion/physical-expr/src/analysis.rs

```diff
@@ -246,3 +246,92 @@ fn calculate_selectivity(
             acc * cardinality_ratio(&initial.interval, &target.interval)
         })
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use arrow_schema::{DataType, Field, Schema};
+    use datafusion::prelude::*;
+    use datafusion_common::{assert_contains, DFSchema};
+    use datafusion_expr::{
+        execution_props::ExecutionProps, interval_arithmetic::Interval,
+    };
+
+    use crate::{create_physical_expr, AnalysisContext};
+
+    use super::{analyze, ExprBoundaries};
+
+    fn make_field(name: &str, data_type: DataType) -> Field {
+        let nullable = false;
+        Field::new(name, data_type, nullable)
+    }
+
+    #[test]
+    fn test_analyze_boundary_exprs() {
+        let schema = Arc::new(Schema::new(vec![make_field("a", DataType::Int64)]));
+        type TestCase = (&'static str, Option<i64>, Option<i64>);
+        let test_cases: Vec<TestCase> = vec![
+            ("a > 10", Some(11), None),
```

Comment:
Another approach is to avoid parsing SQL and insteadl build these expressions programatically 

Like

```suggestion
            (col(a).gt(lit(10), Some(11), None),
```

There are some other examples here: https://docs.rs/datafusion/latest/datafusion/logical_expr/enum.Expr.html#column-references-and-literals

Reply:
I agree, I think it makes more sense to use the expression helpers for two reasons:

* It brings these tests closer to _unit_ tests and further from _integration_ tests, i.e., no need for `SessionContext` and SQL parsing
* It removes the need for the circular dependency

I don't mind changing them over to use the helpers.

Reply:
I did just that in https://github.com/apache/datafusion/commit/a43bf00f6f0e1cf6c521f375859afeb407be4298